### DISCUSSION TITLE
Respect commit template from git configuration

### DISCUSSION THREAD

**Discussion by wl2776** [sourcegit-scm/sourcegit#846] [Q&A]
Git supports [commit templates](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration.html#_commit_template)
Currently SourceGit doesn't respect this setting in favor of its own set of commit templates.

What would be the best way to handle it?

**Comment by wl2776** [comment]
I thinks, it is reasonable to query global setting then all settings for all opened repositories.

**Reply by love-linger** [reply]
`SourceGit` ignores the git's commit templates because you can NOT specify multiple templates to a repository.

**Reply by wl2776** [reply]
Yes, that's right, git config supports only one global messages template and one message template per repository. SourceGit has its own set of templates. 
Do you think it is not reasonable to automatically add  git's templates to SourceGit?

> you can NOT specify multiple templates to a repository

You often emphasize "not" with capital letters. Why?

**Reply by love-linger** [reply]
> You often emphasize "not" with capital letters. Why?

I'm not a native English speaker. This is my personal habit, just to emphasize

**Comment by love-linger** [comment]
I've pushed a new commit that shows the git's commit template in `Template/Histories` popup list.
![image](https://github.com/user-attachments/assets/23d9eb0a-2aa0-4665-8acb-59cf174d6f2b)

> [!NOTE]
> * The commit template configured by git `commit.template` will not be displayed in repository settings panel. You should edit the file directly if you want to modify it.
> * `SourceGit` will use the git's template directly.
